rmf_site_editor/src/interaction/model.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use crate::{interaction::*, site::*};
+use bevy::prelude::*;
+
+pub fn update_model_instance_visual_cues(
+    model_descriptions: Query<
+        (Entity, &Selected, &Hovered),
+        (
+            With<ModelMarker>,
+            With<Group>,
+            Or<(Changed<Hovered>, Changed<Selected>)>,
```

Comment:
In the current setup, we iterate over `&mut model_instances` which does not have this filter, meaning it will include **all** model instances in the world every update cycle. Moreover, this query will **only** include model descriptions whose `Hovered` or `Selected` status has changed during the latest cycle, so [this line](https://github.com/open-rmf/rmf_site/pull/242/files#diff-89f10a882e59e017aa0faa7062467dec447e56c2b827348b80b029fa3668243cR49) will not fetch anything if neither of these components changed for the model description.

I have a feeling this filter was meant to go on `model_instances`, not on `model_descriptions`.

Reply:
My understanding of this implementation is to have the model instances highlighted if their affiliated description is selected/hovered over, rather than the other way round. The current app supports that, such that a change in selection/hovering for a description (queried with the `Changed<>` filter) will trigger an update to all affiliated instances being highlighted (query of all instances in each cycle).

From what I see this behavior also allows relevant task locations to be highlighted when we iterate over every model.